service/history/replicatorQueueProcessor.go

```diff
@@ -173,7 +173,7 @@ func (p *replicatorQueueProcessorImpl) readTasks(readLevel int64) ([]queueTaskIn
 		tasks[i] = response.Tasks[i]
 	}
 
-	return tasks, len(tasks) >= batchSize, nil
+	return tasks, len(response.NextPageToken) != 0, nil
```

Comment:
Looks like you will reissue a new query instead of draining the pagination token.  Can we change that?